### PR TITLE
Fix some insecure issues

### PR DESCRIPTION
**Pull request by nichunen**

### CODE REVIEW

File: server/src/test/java/org/apache/kylin/rest/util/ValidateUtilTest.java [hunk 1]
```diff
@@ -48,6 +48,9 @@
 
     @Test
     public void testCheckIdentifiersExists() throws IOException {
+        String project = "/asd/../_asd33b";
+        project = ValidateUtil.convertStringToBeAlphanumericUnderscore(project);
```
Comment (hit-lacus) [inline server/src/test/java/org/apache/kylin/rest/util/ValidateUtilTest.java]: Could we add some assertEqual for this `project`?

Comment (nichunen) [reply server/src/test/java/org/apache/kylin/rest/util/ValidateUtilTest.java]: @hit-lacus This is some code I submit carelessly, I have removed it

File: core-common/src/main/java/org/apache/kylin/common/persistence/JDBCResourceStore.java [hunk 2]
```diff
@@ -44,28 +45,17 @@
 
 public class JDBCResourceStore extends PushdownResourceStore {
 
-    private static Logger logger = LoggerFactory.getLogger(JDBCResourceStore.class);
-
     public static final String JDBC_SCHEME = "jdbc";
-
+    private static final ConcurrentHashMap<String, Object> resourceHashMap = new ConcurrentHashMap<>();
```
Comment (hit-lacus) [inline core-common/src/main/java/org/apache/kylin/common/persistence/JDBCResourceStore.java]: Maybe change the name `resourceHashMap` to something like `lockObjectMap` is more readable.

Comment (nichunen) [reply core-common/src/main/java/org/apache/kylin/common/persistence/JDBCResourceStore.java]: @hit-lacus Good idea

**Pull request merged**